From 8ca4bdfde63c8fec0ef6d77725589d2bef0774f5 Mon Sep 17 00:00:00 2001
From: Dawei Chen <dchen@silexamerica.com>
Date: Wed, 4 Oct 2017 11:05:37 -0700
Subject: [PATCH] Register attribute handles after GAP and GATT core service
 are registered.  Otherwise it may cause an error message: Failed to obtain
 handles for 'Service Changed' characterisitc.

---
 src/gatt-database.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/src/gatt-database.c b/src/gatt-database.c
index e8ce7d5..caf201c 100644
--- a/src/gatt-database.c
+++ b/src/gatt-database.c
@@ -2422,12 +2422,13 @@ struct btd_gatt_database *btd_gatt_database_new(struct btd_adapter *adapter)
 	database->profiles = queue_new();
 	database->ccc_callbacks = queue_new();
 
+#if 0
 	database->db_id = gatt_db_register(database->db, gatt_db_service_added,
 							gatt_db_service_removed,
 							database, NULL);
 	if (!database->db_id)
 		goto fail;
-
+#endif
 	addr = btd_adapter_get_address(adapter);
 	database->le_io = bt_io_listen(connect_cb, NULL, NULL, NULL, &gerr,
 					BT_IO_OPT_SOURCE_BDADDR, addr,
@@ -2463,6 +2464,12 @@ struct btd_gatt_database *btd_gatt_database_new(struct btd_adapter *adapter)
 
 	register_core_services(database);
 
+	database->db_id = gatt_db_register(database->db, gatt_db_service_added,
+							gatt_db_service_removed,
+							database, NULL);
+	if (!database->db_id)
+		goto fail;
+
 	return database;
 
 fail:
-- 
2.7.4

